stack.go

```diff
@@ -6,47 +6,71 @@ import (
 	"strings"
 )
 
+// Stack is an array of stack frames stored in a human readable format.
+type Stack []StackFrame
+
+// insertFrames inserts a wrap error frame into the correct place of the root error stack trace.
+func (s *Stack) insertFrame(wFrames []StackFrame) {
+	if s == nil || len(wFrames) == 0 {
+		return
+	} else if len(wFrames) == 1 {
+		// append the frame to the end if there's only one
+		*s = append(*s, wFrames[0])
+	}
+
+	for at, f := range *s {
+		if f == wFrames[0] {
+			// return if the stack already contains the frame
+			return
+		} else if f == wFrames[1] {
+			// insert the first frame into the stack if the second frame is found
+			*s = insert(*s, wFrames[0], at)
+			break
```

Comment:
Will there be a case when the second frame is not found and we check for the third one and insert first and second frames into the stack, so and so forth?

Reply:
I don't think so, I think if wFrames[1] is not found, then wFrames[2] and so on will not be in the stack either

Reply:
This case can only come up when wrapping external errors which are basically external root errors. In that case, stack frames will be quite different, but I am assuming that you are not dealing with that case in this PR.

Reply:
In that case, it's sort of an input error and I don't think we can do anything (because we'd have no way of knowing where to insert the external frames afaik). This method handles those cases by basically ignoring junk stacks.